### DISCUSSION TITLE
Preferred way to implement multi-selection.

### DISCUSSION THREAD

**Discussion by nathan-fletcher-seagull-com** [KevinVandy/material-react-table#224] [Q&A]
I'm wondering if there is a preferred way to implement multi-selection when holding down control and shift. I have implemented it myself by hoisting the rowSelection state (and adding my own lastSelectedRow state for shift), and it's working alright, but I want to make sure I'm not missing a better way of doing it.

Note: to avoid clutter I'm not using checkboxes; I'm just implementing selection with an onClick handler on the row like the manual-row-selection-without-checkboxes example.

**Comment by KevinVandy** [comment]
I think that's exactly how I would do it. Feel free to share a stackblitz/code sandbox of your example if you want to share it with others here.
